core/templates/domain/skill/concept-card-backend-api.service.spec.ts

```diff
@@ -127,92 +131,124 @@ describe('Concept card backend API service', function() {
   it('should get all concept cards even the one which was fetched before',
-    function() {
-      $httpBackend.expect('GET', '/concept_card_handler/' + '1').respond(
-        sampleResponse1);
-
-      ConceptCardBackendApiService.loadConceptCards(['1']).then(
-        function(conceptCards) {
-          $httpBackend.expect('GET', '/concept_card_handler/' + '2').respond(
-            sampleResponse2);
-          ConceptCardBackendApiService.loadConceptCards(['1', '2']).then(
-            function(conceptCards2) {
-              expect(conceptCards).toEqual(sampleResponse1.concept_card_dicts);
-              expect(conceptCards2).toEqual(sampleResponse3.concept_card_dicts);
+    fakeAsync(() => {
+      conceptCardBackendApiService.loadConceptCards(['1']).then(
+        (conceptCards) => {
+          conceptCardBackendApiService.loadConceptCards(['1', '2']).then(
+            (conceptCards2) => {
+              expect(conceptCards).toEqual(conceptCardSampleResponse1);
+              expect(conceptCards2).toEqual(conceptCardSampleResponse3);
             });
+
+          var req1 = httpTestingController.expectOne('/concept_card_handler/2');
+          expect(req1.request.method).toEqual('GET');
+          req1.flush(sampleResponse2);
         });
-      $httpBackend.flush();
-    });
+
+      var req2 = httpTestingController.expectOne('/concept_card_handler/1');
+      expect(req2.request.method).toEqual('GET');
+      req2.flush(sampleResponse1);
+
+      flushMicrotasks();
+    }));
 
   it('should use the rejection handler if backend request failed',
-    function() {
-      var successHandler = jasmine.createSpy('success');
-      var failHandler = jasmine.createSpy('fail');
+    fakeAsync(() => {
+      let successHandler = jasmine.createSpy('success');
+      let failHandler = jasmine.createSpy('fail');
 
-      $httpBackend.expect('GET', '/concept_card_handler/1').respond(
-        500, 'Error loading skill 1.');
-      ConceptCardBackendApiService.loadConceptCards(['1']).then(
+      conceptCardBackendApiService.loadConceptCards(['1']).then(
         successHandler, failHandler);
-      $httpBackend.flush();
+      var req = httpTestingController.expectOne('/concept_card_handler/1');
+      expect(req.request.method).toEqual('GET');
+      req.flush(null, {
+        status: 500, statusText: 'Error loading skill 1.'
+      });
 
-      expect(successHandler).not.toHaveBeenCalled();
-      expect(failHandler).toHaveBeenCalledWith('Error loading skill 1.');
-    });
+      flushMicrotasks();
 
-  it('should not fetch the same concept card', function() {
-    $httpBackend.expect('GET', '/concept_card_handler/1').respond(
-      sampleResponse1);
-    ConceptCardBackendApiService.loadConceptCards(['1'])
-      .then(function(conceptCards) {
-        ConceptCardBackendApiService.loadConceptCards(['1'])
-          .then(function(conceptCards2) {
+      expect(successHandler).not.toHaveBeenCalled();
+      expect(failHandler).toHaveBeenCalled();
```

Comment:
Should this be `toHaveBeenCalledWith`?

Reply:
Calling this as `toHaveBeenCalledWith` causes the tests to fail with the following error:
```
Expected spy fail to have been called with:
	  [ 'Error loading skill 1.' ]
	but actual calls were:
	  [ null ].
```

I referenced [this PR](https://github.com/oppia/oppia/pull/8501/files#diff-049e43d749ba9e37f2869f814e2c3381R104) when deciding to switch from `toHaveBeenCalledWith` to `toHaveBeenCalled`.

Reply:
Interesting, did you look into why that happens? I think it may be because we pass null as the first param to the flush function in line 203. Can you try passing the error string instead?

See example here:
https://github.com/oppia/oppia/blob/3b8f43fe674205aa579f973849adf28c30b85c52/core/templates/domain/collection/editable-collection-backend-api.service.spec.ts#L88-L95

Reply:
Ah, that makes sense. Fixed now!